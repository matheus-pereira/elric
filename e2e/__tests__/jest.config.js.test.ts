/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as path from 'path';
import {wrap} from 'elric-snapshot-serializer-raw';
import {cleanup, extractSummary, writeFiles} from '../Utils';
import runelric from '../runelric';

const DIR = path.resolve(__dirname, '../elric-config-js');

beforeEach(() => cleanup(DIR));
afterAll(() => cleanup(DIR));

test('works with elric.config.js', () => {
  writeFiles(DIR, {
    '__tests__/a-banana.js': `test('banana', () => expect(1).toBe(1));`,
    'elric.config.js': `module.exports = {testRegex: '.*-banana.js'};`,
    'package.json': '{}',
  });

  const {stderr, exitCode} = runelric(DIR, ['-w=1', '--ci=false']);
  const {rest, summary} = extractSummary(stderr);
  expect(exitCode).toBe(0);
  expect(wrap(rest)).toMatchSnapshot();
  expect(wrap(summary)).toMatchSnapshot();
});

test('traverses directory tree up until it finds elric.config', () => {
  writeFiles(DIR, {
    '__tests__/a-banana.js': `
    const slash = require('slash');
    test('banana', () => expect(1).toBe(1));
    test('abc', () => console.log(slash(process.cwd())));
    `,
    'elric.config.js': `module.exports = {testRegex: '.*-banana.js'};`,
    'package.json': '{}',
    'some/nested/directory/file.js': '// nothing special',
  });

  const {stderr, exitCode, stdout} = runelric(
    path.join(DIR, 'some', 'nested', 'directory'),
    ['-w=1', '--ci=false'],
    {skipPkgJsonCheck: true},
  );

  // Snapshot the console.loged `process.cwd()` and make sure it stays the same
  expect(
    wrap(stdout.replace(/^\W+(.*)e2e/gm, '<<REPLACED>>')),
  ).toMatchSnapshot();

  const {rest, summary} = extractSummary(stderr);
  expect(exitCode).toBe(0);
  expect(wrap(rest)).toMatchSnapshot();
  expect(wrap(summary)).toMatchSnapshot();
});

test('invalid JS in elric.config.js', () => {
  writeFiles(DIR, {
    '__tests__/a-banana.js': `test('banana', () => expect(1).toBe(1));`,
    'elric.config.js': `module.exports = i'll break this file yo`,
    'package.json': '{}',
  });

  const {stderr, exitCode} = runelric(DIR, ['-w=1', '--ci=false']);
  expect(stderr).toMatch('SyntaxError: ');
  expect(exitCode).toBe(1);
});
