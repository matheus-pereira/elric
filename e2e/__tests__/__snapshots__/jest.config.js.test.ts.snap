// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`traverses directory tree up until it finds elric.config 1`] = `
  console.log
<<REPLACED>>/elric-config-js/some/nested/directory

      at Object.log (__tests__/a-banana.js:3:27)

`;

exports[`traverses directory tree up until it finds elric.config 2`] = `
PASS ../../../__tests__/a-banana.js
  ✓ banana
  ✓ abc
`;

exports[`traverses directory tree up until it finds elric.config 3`] = `
Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
`;

exports[`works with elric.config.js 1`] = `
PASS __tests__/a-banana.js
  ✓ banana
`;

exports[`works with elric.config.js 2`] = `
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
`;
